### DISCUSSION TITLE
Request for Feedback: Which language for a new Official Starter Snake?

### DISCUSSION THREAD

**Discussion by bvanvugt** [BattlesnakeOfficial/feedback#79] [Announcements & News]
We're looking to revise (and likely add to) the set of [Official Battlesnake Starter Projects](https://docs.battlesnake.com/references/starter-projects) and we wanted to ask the community what language(s) are popular, folks are looking to learn and experiment with, and could use some additional support?

**Some contenders to start the discussion:**
- Rust
- Kotlin
- TypeScript

**Comment by InterestingBrainPoops** [comment]
I cant speak to Kotlin or TypeScript, but making a starter snake for rust would be complicated.
There are many libraries for this kind of thing, some more complicated than others. 
If you want simple and easy to read/use, id reccomend using warp + serde. It provides a simple way of reading in the json, and also has an express-like web server. Its also asynchronous by default. The [warpy-snake](https://github.com/Osrepnay/warpy-snake) template is a great example of what it would look like.
Another high level library would be rocket. I havent had much experience with it, but from the looks of it, it's very easy to use.
Theres also the option of handrolling it, or using lower level libraries.

**Reply by coreyja** [reply]
> Another high level library would be rocket. I havent had much experience with it, but from the looks of it, it's very easy to use.

My snakes use Rocket in local dev. I like Rocket cause its pretty 'full-featured' as far as web frameworks go in Rust. Which is good if you want to explore more things after Battlesnake, but I do feel like some of its functionality you just don't _need_ in Battlesnake 🤷 

Here is my rocket source code for reference: https://github.com/coreyja/battlesnake-rs/blob/main/web-rocket/src/main.rs

**Comment by bvanvugt** [comment]
We've since added official support for both Rust and TypeScript 👍
